src/connectors/multi-index-results/connectMultiIndexResults.js

```diff
@@ -0,0 +1,112 @@
+import find from 'lodash/find';
+
+import escapeHits, { tagConfig } from '../../lib/escape-highlight';
+import { checkRendering } from '../../lib/utils';
+
+const usage = `Usage:
+var customMultiIndexResults = connectMultiIndexResults(function render(params, isFirstRendering) {
+  // params = {
+  //   derivedIndices,
+  //   instantSearchInstance,
+  //   widgetParams
+  // }
+});
+search.addWidget(
+  customMultiIndexResults({
+    indices,
+    [ escapeHits = false ]
+  })
+);
+Full documentation available at https://community.algolia.com/instantsearch.js/connectors/connectMultiIndexResults.html
+`;
+
+export default function connectMultiIndexResults(renderFn, unmountFn) {
+  checkRendering(renderFn, usage);
+
+  return (widgetParams = {}) => {
+    const { indices } = widgetParams;
+
+    if (
+      !indices ||
+      !Array.isArray(indices) ||
+      (Array.isArray(indices) && indices.length < 1)
+    ) {
+      throw new Error(usage);
+    }
+
+    return {
+      getConfiguration() {
+        return widgetParams.escapeHits ? tagConfig : undefined;
+      },
+
+      init({ instantSearchInstance, helper }) {
+        // init `this.derivedIndices` with the default index
+        this.derivedIndices = [
+          {
+            helper,
+            label: 'default',
+            index: helper.getIndex(),
+            results: undefined,
+          },
+        ];
+
+        // add additionnal indices intto `this.derivedIndices`
+        indices.forEach(({ label, value }) => {
+          const derivedHelper = helper.derive(searchParameters =>
+            searchParameters.setIndex(value)
+          );
+          this.derivedIndices.push({
+            label,
+            index: value,
+            helper: derivedHelper,
+            results: undefined,
+          });
+
+          // after a derived index got results, update `this.derivedIndices[x].results`
+          // and trigger a new render
+          derivedHelper.on('result', results =>
+            this.saveResults({ results, index: value })
+          );
+        });
+
+        this.instantSearchInstance = instantSearchInstance;
+        this.renderWithAllIndices({ isFirstRendering: true });
+      },
+
+      saveResults({ results, index }) {
+        const derivedIndice = find(this.derivedIndices, { index });
+
+        if (
+          widgetParams.escapeHits &&
+          results.hits &&
+          results.hits.length > 0
+        ) {
+          results.hits = escapeHits(results.hits);
+        }
+
+        derivedIndice.results = results;
+        this.renderWithAllIndices();
```

Comment:
I don't follow how you actually saved the results 🤔

Reply:
* Every index is stored into an object on the connector instance

```
{ label, index, helper, results }
```

* On every search, I update the `results` and trigger a new render 👍

Reply:
would be nice to leave a comment for the behaviour of doing side effects on render